os/drivers/lcd/mipi_lcd.c

```diff
@@ -169,12 +174,35 @@ static int lcd_putrun(FAR struct lcd_dev_s *dev, fb_coord_t row, fb_coord_t col,
 static int lcd_putarea(FAR struct lcd_dev_s *dev, fb_coord_t row_start, fb_coord_t row_end, fb_coord_t col_start, fb_coord_t col_end, FAR const uint8_t *buffer, fb_coord_t stride)
 {
 	FAR struct mipi_lcd_dev_s *priv = (FAR struct mipi_lcd_dev_s *)dev;
+
+	uint8_t *final_buffer = buffer;
+#ifdef CONFIG_LCD_ST7701SN
+	int src_buf_ind = 0;
+	int dest_buf_ind = 0;
+	if(is_lcd_buffer_1) {
+		final_buffer = lcd_buffer_2;
+		is_lcd_buffer_1 = false;
+	}
+	else {
+		final_buffer = lcd_buffer_1;
+		is_lcd_buffer_1 = true;
+	}
+
+	for(int src_row = 0; src_row < CONFIG_LCD_YRES; src_row++) {
+		for(int src_col = CONFIG_LCD_XRES - 1; src_col >= 0 ; src_col--) {
+			src_buf_ind = ((CONFIG_LCD_XRES * src_col) + src_row) * 2;
+			final_buffer[dest_buf_ind] = buffer[src_buf_ind];
+			final_buffer[dest_buf_ind+1] = buffer[src_buf_ind+1];
+			dest_buf_ind += 2;
```

Comment:
This seems to assume the resolution type used is always RGB565. As I knew, ST7701SN seems using higher resolution such as RGB888, then each pixel will be 3bytes.
Please check whether some macro should be used to differentiate that.

Reply:
Need to check